src/logging.cpp

```diff
@@ -334,19 +339,20 @@ namespace BCLog {
     }
 } // namespace BCLog
 
-void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line, const BCLog::LogFlags category, const BCLog::Level level)
+void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const SourceLocation& source_location, const BCLog::LogFlags category, const BCLog::Level level)
 {
     StdLockGuard scoped_lock(m_cs);
     std::string str_prefixed = LogEscapeMessage(str);
 
-    if ((category != LogFlags::NONE || level != Level::None) && m_started_new_line) {
+    const bool print_category{category != LogFlags::NONE && category != LogFlags::UNCONDITIONAL_ALWAYS && category != LogFlags::UNCONDITIONAL_RATE_LIMITED};
```

Comment:
Is the `category != LogFlags::NONE` necessary here? I don't think anything logs with category NONE now.

Reply:
There are still tests that log using NONE:
https://github.com/bitcoin/bitcoin/blob/a6377c5fe4f8d8fabbae527df888afdd9c6f771a/src/test/logging_tests.cpp#L80-L81

What do you think of removing the NONE category entirely?

Reply:
I think those should be replaced with `LogPrintf_(..., BCLog::LogFlags::UNCONDITIONAL_ALWAYS, ...)`, since it's impossible for `LogPrintf_(..., BCLog::LogFlags::NONE, ...)` to be called in the produce code.

It seems fine to remove `BCLog::LogFlags::NONE` after that, but I don't think it's necessary in this PR.